hnn_core/opt_toy_example/general.py

```diff
@@ -0,0 +1,344 @@
+import numpy as np
+import matplotlib.pyplot as plt 
+from collections import namedtuple
+
+from hnn_core import read_params, jones_2009_model, simulate_dipole 
+from hnn_core.network import pick_connection
+
+from skopt import gp_minimize 
+from skopt.utils import use_named_args
+from skopt.space import Real
+
+from scipy.optimize import fmin_cobyla
+
+class Optimizer:
+    def __init__(self, net, constraints, solver, metric):
+        self.net = net
+        self.constraints = constraints
+        if solver == 'bayesian':
+            self._get_params = _get_params_bayesian
+            self._run_opt = _run_opt_bayesian
+        elif solver == 'cobyla':
+            self._get_params = _get_params_cobyla
+            self._run_opt = _run_opt_cobyla
+        if metric == 'evoked':
+            self.metric = _rmse_evoked
+        elif metric == 'rhythmic':
+            self.metric = _rmse_rhythmic
+        self.net_ = None
+        self.error_values = None
+        self.opt_params = None
+        
+    def fit(self, target_statistic, window_len = None):
+        """ ...
+            Parameters
+            ----------
+            target_statistic : dpl
+            window_len : for smoothing dpl
```

Comment:
I think `window_len` should be specified somehow with the `metric` or possibly as an argument of the objective function since it is part of the cost calculation.